### PR TITLE
[W5.4][T17-A2]Khoo Wei Ping

### PR DESCRIPTION
**Pull request by khooroko**
Include sort command and functionality, updated I/O tests, JUnit tests and user guide accordingly

### CODE REVIEW

**Comment by Gideonfu** [comment]
Hi Wei Ping, good job on adding a Sort command to AddressBook.

File: src/seedu/addressbook/data/AddressBook.java [hunk 1]
```diff
@@ -13,12 +8,17 @@
 import seedu.addressbook.data.tag.Tag;
 import seedu.addressbook.data.tag.UniqueTagList;
 
+import java.util.HashMap;
```
Comment (ang-zy) [inline src/seedu/addressbook/data/AddressBook.java]: Coding convention is to have imports listed in alphabetical order

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 2]
```diff
@@ -94,4 +96,21 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+
+    /**
+     * Comparator for comparing persons.
+     */
+    Comparator<ReadOnlyPerson> READ_ONLY_PERSON_COMPARATOR = new Comparator<ReadOnlyPerson>() {
```
Comment (ang-zy) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java]: Would making ReadOnlyPerson implement _Comparable_ be a simpler method?

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 3]
```diff
@@ -100,8 +101,8 @@ public boolean contains(ReadOnlyPerson toCheck) {
      * Adds a person to the list.
      *
      * @throws DuplicatePersonException if the person to add is a duplicate of an existing person in the list.
-     *    The @link{ReadOnlyPerson#isSamePerson} method is used for this comparison,
-     *    which defines a weaker notion of equality.
+     *                                  The @link{ReadOnlyPerson#isSamePerson} method is used for this comparison,
+     *                                  which defines a weaker notion of equality.
```
Comment (ang-zy) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: The indentation here isn't really necessary

**Pull request closed without merging**